src/Product/ProductMetaHandler.php

```diff
@@ -265,7 +265,7 @@ protected function sanitize_meta_query( array $queries ): array {
 	 *
 	 * @return array
 	 */
-	public function prefix_meta_query_keys( array $meta_queries ): array {
+	public function prefix_meta_query_keys( $meta_queries ): array {
```

Comment:
If we take away the requirement for the `meta_query` to be an array then the doc block should be updated as well. Although it's not quite clear why this is needed. Wouldn't it be better to check if it's an array before assuming it's a nested meta_query? If we do then it seems like we would always pass an array to the function, WP says `meta_query` should be one or multiple array's.
Same logic for the `sanitize_meta_query` function.

Reply:
> Wouldn't it be better to check if it's an array before assuming it's a nested meta_query?

The first check for `is_array` ensures that the `else` at the end is always running for a nested array query, since anything but an array won't get there.

> If we do then it seems like we would always pass an array to the function, WP says meta_query should be one or multiple arrays.

This part of the code is based on the `WP_Meta_Query::sanitize_query` method from WP core with some modifications to validate the meta keys and prefix them.

This `is_array` check is working as a soft fail-switch that sanitizes the meta query but also doesn't throw any exceptions. I assume this is a backward compatibility measure taken by WP core, and since the `sanitize_meta_query` is run on all product queries (via hooking to `woocommerce_product_data_store_cpt_get_products_query`) I think it'd be safer to just treat the meta queries the same as WP core.

> If we take away the requirement for the meta_query to be an array then the doc block should be updated as well. 
As for the doc block, It'd be best to leave that as expecting an `array` because we actually expect one here (similar to `WP_Meta_Query::__construct` and `WP_Meta_Query::sanitize_query`). The `is_array` check passes and the method runs only if the input is an array.

Reply:
Alright, sounds good. It was working fine, just thought it was an odd pattern to assume it's recursive and then bail out within the function if it's not. Happy to leave it based off of the WP function though.